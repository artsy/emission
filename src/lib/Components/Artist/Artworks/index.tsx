import * as React from "react"
import { StyleSheet, View, ViewProperties } from "react-native"
import * as Relay from "react-relay"

import ArtworksGrid from "../../artwork_grids/relay_connections/artist_artworks_grid"
import Separator from "../../separator"
import SerifText from "../../text/serif"

import colors from "../../../../data/colors"

interface Props extends ViewProperties {
  artist: {
    counts: {
      for_sale_artworks: number
      artworks: number
    }
    not_for_sale_artworks: any[]
    for_sale_artworks: any[]
  }
  relay: any
}

interface State {
  completedForSaleWorks: boolean
}

class Artworks extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      completedForSaleWorks: false,
    }
  }

  render() {
    const forSaleCount = this.props.artist.counts.for_sale_artworks
    const otherCount = this.props.artist.counts.artworks - forSaleCount
    if (forSaleCount === 0) {
      return this.renderSection({
        title: "Works",
        count: otherCount,
        filter: "IS_NOT_FOR_SALE",
        onComplete: null,
      })
    } else {
      const otherWorks: any[] = []
      const showOtherWorks = otherCount > 0 && (forSaleCount < 10 || this.state.completedForSaleWorks)
      if (showOtherWorks) {
        otherWorks.push(<Separator style={styles.sectionSeparator} key="separator" />)
        otherWorks.push(
          this.renderSection({
            title: "Other Works",
            count: otherCount,
            filter: "IS_NOT_FOR_SALE",
            onComplete: null,
          })
        )
      }
      return (
        <View style={styles.section}>
          {this.renderSection({
            title: "Works for Sale",
            count: forSaleCount,
            filter: "IS_FOR_SALE",
            onComplete: () => {
              this.setState({ completedForSaleWorks: true })
            },
          })}
          {otherWorks}
        </View>
      )
    }
  }

  renderSection({ title, count, filter, onComplete }) {
    const countStyles = [styles.text, styles.count]
    return (
      <View key={title}>
        <SerifText style={styles.heading}>
          <SerifText style={styles.text}>{title}</SerifText> <SerifText style={countStyles}>({count})</SerifText>
        </SerifText>
        <ArtworksGrid artist={this.props.artist} filter={filter} onComplete={onComplete} queryKey="artist" />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  heading: {
    marginBottom: 20,
  },
  text: {
    fontSize: 20,
  },
  count: {
    color: colors["gray-semibold"],
  },
  section: {
    marginBottom: 40,
  },
  sectionSeparator: {
    marginTop: 40, // FIXME: This is because the above `section.marginBottom` didnâ€™t work before the separator.
    marginBottom: 20,
  },
})

export default Relay.createContainer(Artworks, {
  fragments: {
    artist: () => Relay.QL`
      fragment on Artist {
        counts {
          artworks
          for_sale_artworks
        }
        ${ArtworksGrid.getFragment("artist", { filter: "IS_FOR_SALE" })}
        ${ArtworksGrid.getFragment("artist", { filter: "IS_NOT_FOR_SALE" })}
      }
    `,
  },
})

interface RelayProps {
  artist: {
    counts: {
      artworks: boolean | number | string | null
      for_sale_artworks: boolean | number | string | null
    } | null
    for_sale_artworks: Array<boolean | number | string | null> | null
    not_for_sale_artworks: Array<boolean | number | string | null> | null
  }
}
