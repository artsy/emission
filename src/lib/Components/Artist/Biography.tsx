import * as React from "react"
import { Dimensions, StyleSheet, View, ViewProperties } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"

import removeMarkdown from "remove-markdown"

import Headline from "../Text/Headline"
import SerifText from "../Text/Serif"

const sideMargin = Dimensions.get("window").width > 700 ? 50 : 0

interface Props extends ViewProperties {
  artist: any
}

class Biography extends React.Component<Props, any> {
  render() {
    const artist = this.props.artist
    if (!artist.blurb && !artist.bio) {
      return null
    }

    return (
      <View style={{ marginLeft: sideMargin, marginRight: sideMargin }}>
        <Headline style={{ marginBottom: 20 }}>Biography</Headline>
        {this.blurb(artist)}
        <SerifText style={styles.bio} numberOfLines={0}>
          {this.bioText()}
        </SerifText>
      </View>
    )
  }

  blurb(artist) {
    if (artist.blurb) {
      return (
        <SerifText style={styles.blurb} numberOfLines={0}>
          {removeMarkdown(artist.blurb)}
        </SerifText>
      )
    }
  }

  bioText() {
    const bio = this.props.artist.bio
    return bio.replace("born", "b.")
  }
}

const styles = StyleSheet.create({
  blurb: {
    fontSize: 16,
    lineHeight: 20,
    marginBottom: 20,
  },
  bio: {
    fontSize: 16,
    lineHeight: 20,
    marginBottom: 40,
  },
})

export default createFragmentContainer(
  Biography,
  graphql`
    fragment Biography_artist on Artist {
      bio
      blurb
    }
  `
)

interface RelayProps {
  artist: {
    bio: string | null
    blurb: string | null
  }
}
