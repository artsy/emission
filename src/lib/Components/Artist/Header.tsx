import React from "react"
import { Dimensions, NativeModules, StyleSheet, TextStyle, View, ViewStyle } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import { Schema, Track, track as _track } from "../../utils/track"
const { ARTemporaryAPIModule } = NativeModules
import { Button } from "@artsy/palette"
import colors from "lib/data/colors"
import Headline from "../Text/Headline"
import SerifText from "../Text/Serif"

import { Header_artist } from "__generated__/Header_artist.graphql"

const isPad = Dimensions.get("window").width > 700

interface Props {
  artist: Header_artist
}

interface State {
  following: boolean
  followersCount: number
}

const track: Track<Props, State> = _track

@track()
class Header extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      following: false,
      followersCount: props.artist.counts.follows as number,
    }
  }

  componentDidMount() {
    NativeModules.ARTemporaryAPIModule.followStatusForArtist(this.props.artist.internalID, (error, following) => {
      if (error) {
        // FIXME: Handle error
        console.error("Artist/Header.tsx", error.message)
      }
      this.setState({ following })
    })
  }

  render() {
    const artist = this.props.artist
    return (
      <View style={{ paddingTop: 20 }}>
        <Headline style={[styles.base, styles.headline]}>{artist.name}</Headline>
        {this.renderByline()}
        {this.renderFollowersCount()}
        {this.renderFollowButton()}
      </View>
    )
  }

  renderFollowButton() {
    if (this.state.following !== null) {
      return (
        <View style={styles.followButton}>
          <Button
            variant={this.state.following ? "secondaryOutline" : "primaryBlack"}
            block
            width={100}
            onPress={this.handleFollowChange.bind(this)}
          >
            {this.state.following ? "Following" : "Follow"}
          </Button>
        </View>
      )
    }
  }

  renderFollowersCount() {
    const count = this.state.followersCount
    const followerString = count + (count === 1 ? " Follower" : " Followers")
    return <SerifText style={[styles.base, styles.followCount]}>{followerString}</SerifText>
  }

  renderByline() {
    const artist = this.props.artist
    const bylineRequired = artist.nationality || artist.birthday
    if (bylineRequired) {
      return (
        <View>
          <SerifText style={styles.base}>{this.descriptiveString()}</SerifText>
        </View>
      )
    } else {
      return null
    }
  }

  descriptiveString() {
    const artist = this.props.artist
    const descriptiveString = (artist.nationality || "") + this.birthdayString()
    return descriptiveString
  }

  birthdayString() {
    const birthday = this.props.artist.birthday
    if (!birthday) {
      return ""
    }

    const leadingSubstring = this.props.artist.nationality ? ", b." : ""

    if (birthday.includes("born")) {
      return birthday.replace("born", leadingSubstring)
    } else if (birthday.includes("Est.") || birthday.includes("Founded")) {
      return " " + birthday
    }

    return leadingSubstring + " " + birthday
  }

  @track((props, state) => ({
    action_name: state.following ? Schema.ActionNames.ArtistUnfollow : Schema.ActionNames.ArtistFollow,
    action_type: Schema.ActionTypes.Tap,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.gravityID,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  handleFollowChange() {
    const newFollowersCount = this.state.following ? this.state.followersCount - 1 : this.state.followersCount + 1
    ARTemporaryAPIModule.setFollowArtistStatus(
      !this.state.following,
      this.props.artist.internalID,
      (error, following) => {
        if (error) {
          console.warn(error)
          this.failedFollowChange()
        } else {
          this.successfulFollowChange()
        }
        this.setState({ following, followersCount: newFollowersCount })
      }
    )
    this.setState({ following: !this.state.following, followersCount: newFollowersCount })
  }

  @track((props, state) => ({
    action_name: state.following ? Schema.ActionNames.ArtistFollow : Schema.ActionNames.ArtistUnfollow,
    action_type: Schema.ActionTypes.Success,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.gravityID,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  successfulFollowChange() {
    // callback for analytics purposes
  }

  @track((props, state) => ({
    action_name: state.following ? Schema.ActionNames.ArtistFollow : Schema.ActionNames.ArtistUnfollow,
    action_type: Schema.ActionTypes.Fail,
    owner_id: props.artist.internalID,
    owner_slug: props.artist.gravityID,
    owner_type: Schema.OwnerEntityTypes.Artist,
  }))
  failedFollowChange() {
    // callback for analytics purposes
  }
}

interface Styles {
  base: TextStyle
  headline: TextStyle
  followCount: TextStyle
  followButton: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  base: {
    textAlign: "center",
  },
  headline: {
    fontSize: 14,
  },
  followCount: {
    color: colors["gray-semibold"],
    marginBottom: 30,
  },
  followButton: {
    height: 40,
    width: isPad ? 330 : null,
    alignSelf: isPad ? "center" : null,
    marginLeft: 0,
    marginRight: 0,
  },
})

export default createFragmentContainer(Header, {
  artist: graphql`
    fragment Header_artist on Artist {
      internalID
      slug
      name
      nationality
      birthday
      counts {
        follows
      }
    }
  `,
})
