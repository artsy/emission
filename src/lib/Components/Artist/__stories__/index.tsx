import "./ArtistArticles.story"
import "./ArtistHeader.story"
