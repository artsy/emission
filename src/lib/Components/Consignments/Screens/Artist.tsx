import * as React from "react"

import ArtistSearch from "../Components/ArtistSearchResults"
import DoneButton from "../Components/BottomAlignedButton"
import { ConsignmentSetup } from "../index"

import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"

interface Props extends ConsignmentSetup {
  navigator: NavigatorIOS
  route: Route
}

export default class Artist extends React.Component<Props, null> {
  doneTapped = () => {
    // Update state
    this.props.navigator.pop()
  }

  render() {
    return (
      <View style={{ backgroundColor: "black", flex: 1 }}>
        <DoneButton onPress={this.doneTapped}>
          <View
            style={{ alignContent: "center", justifyContent: "center", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <ArtistSearch results={null} query={null} />
          </View>
        </DoneButton>
      </View>
    )
  }
}
