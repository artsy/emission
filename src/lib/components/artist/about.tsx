import * as Relay from 'react-relay'
import * as React from 'react'

import { View, StyleSheet, Dimensions, ViewProperties } from 'react-native'

import Biography from './biography'
import Articles from './articles'
import RelatedArtists from '../related_artists'
import Separator from '../separator'
import NavButton from '../buttons/navigation_button'

interface Props extends ViewProperties {
  artist: any
}

class About extends React.Component<Props, null> {
  render() {
    return (
      <View>
        { this.biography() }
        { this.articles() }
        { this.relatedArtists() }
      </View>
    )
  }

  biography() {
    if (this.props.artist.has_metadata) {
      return (
        <View>
          <Biography artist={this.props.artist} />
          { this.auctionResults() }
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  auctionResults() {
    if (this.props.artist.is_display_auction_link) {
      // Keeps the same margins as the bio text
      const sideMargin = Dimensions.get('window').width > 700 ? 50 : 20
      const url = `/artist/${this.props.artist.id}/auction-results`
      return <NavButton title="Auction Results" href={url} style={{ marginLeft: sideMargin, marginRight: sideMargin }}/>
    }
  }

  articles() {
    if (this.props.artist.articles.length) {
      return (
        <View>
          <Articles articles={this.props.artist.articles} />
          <Separator style={styles.sectionSeparator} />
        </View>
      )
    }
  }

  relatedArtists() {
    return this.props.artist.related_artists.length
      ? <RelatedArtists artists={this.props.artist.related_artists}/>
      : null
  }

}

const styles = StyleSheet.create({
  sectionSeparator: {
    marginBottom: 20,
  }
})

export default Relay.createContainer(About, {
  fragments: {
    artist: () => Relay.QL`
      fragment on Artist {
        has_metadata
        is_display_auction_link
        id
        ${Biography.getFragment('artist')}
        related_artists: artists(size: 16) {
          ${RelatedArtists.getFragment('artists')}
        }
        articles {
          ${Articles.getFragment('articles')}
        }
      }
    `,
  }
})
