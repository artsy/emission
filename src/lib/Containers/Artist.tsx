// import "@babel/runtime"

import React, { Component } from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { Dimensions, ScrollView, StyleSheet, View, ViewProperties, ViewStyle } from "react-native"

import About from "lib/Components/Artist/About"
import Artworks from "lib/Components/Artist/Artworks"
import Header from "lib/Components/Artist/Header"
import Shows from "lib/Components/Artist/Shows"

import { SwitchEvent } from "lib/Components/SwitchView"
import TabView from "lib/Components/TabView"
import { Schema, Track, track as _track } from "lib/utils/track"

import { Artist_artist } from "__generated__/Artist_artist.graphql"

const isPad = Dimensions.get("window").width > 700

// jest.unmock("lib/utils/track")

const TABS = {
  ABOUT: "ABOUT",
  WORKS: "WORKS",
  SHOWS: "SHOWS",
}

interface Props extends ViewProperties {
  artist: Artist_artist
}

interface State {
  selectedTabIndex: number
  selectedTabTitle: string
}

const track: Track<Props, State> = _track

@track()
export class Artist extends Component<Props, State> {
  initialTabState() {
    const tabs = this.availableTabs()
    const worksTab = tabs.indexOf(TABS.WORKS)
    if (worksTab > -1) {
      return { selectedTabIndex: worksTab, selectedTabTitle: TABS.WORKS }
    } else {
      return { selectedTabIndex: 0, selectedTabTitle: tabs[0] }
    }
  }

  componentWillMount() {
    this.setState(this.initialTabState())
  }

  @track((props, state) => {
    let actionName
    switch (state.selectedTabTitle) {
      case TABS.ABOUT:
        actionName = Schema.ActionNames.ArtistAbout
        break
      case TABS.WORKS:
        actionName = Schema.ActionNames.ArtistWorks
        break
      case TABS.SHOWS:
        actionName = Schema.ActionNames.ArtistShows
        break
    }
    return {
      action_name: actionName,
      action_type: Schema.ActionTypes.Tap,
      owner_id: props.artist._id,
      owner_slug: props.artist.id,
      owner_type: Schema.OwnerEntityTypes.Artist,
    }
  })
  tabSelectionDidChange(event: SwitchEvent) {
    this.setState({ selectedTabIndex: event.nativeEvent.selectedIndex, selectedTabTitle: this.selectedTabTitle() })
  }

  availableTabs = () => {
    const tabs: string[] = []
    const artist = this.props.artist
    const displayAboutSection = artist.has_metadata || artist.counts.articles > 0 || artist.counts.related_artists > 0

    if (displayAboutSection) {
      tabs.push(TABS.ABOUT)
    }

    if (artist.counts.artworks) {
      tabs.push(TABS.WORKS)
    }

    if (artist.counts.partner_shows) {
      tabs.push(TABS.SHOWS)
    }
    return tabs
  }

  selectedTabTitle = () => {
    return this.availableTabs()[this.state.selectedTabIndex]
  }

  renderSelectedTab = () => {
    switch (this.selectedTabTitle()) {
      case TABS.ABOUT:
        return <About artist={this.props.artist as any} />
      case TABS.WORKS:
        return <Artworks artist={this.props.artist as any} />
      case TABS.SHOWS:
        return <Shows artist={this.props.artist as any} />
    }
  }

  renderTabView() {
    return (
      <TabView
        titles={this.availableTabs()}
        selectedIndex={this.state.selectedTabIndex}
        onSelectionChange={this.tabSelectionDidChange.bind(this)}
        style={styles.tabView}
      >
        {this.renderSelectedTab()}
      </TabView>
    )
  }

  renderSingleTab() {
    return <View style={{ paddingTop: 30 }}>{this.renderSelectedTab()}</View>
  }

  render() {
    const windowDimensions = Dimensions.get("window")
    const commonPadding = windowDimensions.width > 700 ? 40 : 20
    const displayTabView = this.availableTabs().length > 1

    return (
      <ScrollView scrollsToTop={true} automaticallyAdjustContentInsets={false}>
        <View style={{ paddingLeft: commonPadding, paddingRight: commonPadding }}>
          <Header artist={this.props.artist as any} />
          {displayTabView ? this.renderTabView() : this.renderSingleTab()}
        </View>
      </ScrollView>
    )
  }
}

interface Styles {
  tabView: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  tabView: {
    width: isPad ? 330 : null,
    marginTop: 30,
    marginBottom: 30,
    alignSelf: isPad ? "center" : null,
  },
})

export default createFragmentContainer(
  Artist,
  graphql`
    fragment Artist_artist on Artist {
      _id
      id
      has_metadata
      counts {
        artworks
        partner_shows
        related_artists
        articles
      }
      ...Header_artist
      ...About_artist
      ...Shows_artist
      ...Artworks_artist
    }
  `
)
