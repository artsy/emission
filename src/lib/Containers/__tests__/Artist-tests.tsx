import React from "react"
import "react-native"
import * as renderer from "react-test-renderer"

import { Artist } from "../Artist"

describe("availableTabs", () => {
  it("returns nothing if artist has no metadata, shows, or works", () => {
    const artist = new Artist(artistProps(false))
    expect(artist.availableTabs()).toEqual([])
  })

  it("returns About tab if artist has metadata", () => {
    const artist = new Artist(artistProps(true))
    expect(artist.availableTabs()).toEqual(["ABOUT"])
  })

  it("returns About tab if artist has articles", () => {
    const artist = new Artist(artistProps(false, { articles: 1 }))
    expect(artist.availableTabs()).toEqual(["ABOUT"])
  })

  it("returns Shows tab if artist has shows", () => {
    const artist = new Artist(artistProps(false, { partner_shows: 2 }))
    expect(artist.availableTabs()).toEqual(["SHOWS"])
  })

  it("returns Works tab if artist has works", () => {
    const artist = new Artist(artistProps(false, { artworks: 2 }))
    expect(artist.availableTabs()).toEqual(["WORKS"])
  })

  it("returns all three tabs if artist has metadata, works, and shows", () => {
    const artist = new Artist(artistProps(true, { artworks: 1, partner_shows: 1 }))
    expect(artist.availableTabs()).toEqual(["ABOUT", "WORKS", "SHOWS"])
  })
})

describe("after rendering", () => {
  it("mounts with Works tab selected if works exist", () => {
    const artist = new Artist(artistProps(false, { artworks: 5 }))
    const worksTabIndex = artist.availableTabs().indexOf("WORKS")
    expect(artist.initialTabState()).toEqual({ selectedTabIndex: worksTabIndex, selectedTabTitle: "WORKS" })
  })

  it("mounts at the first tab index if artist has no works", () => {
    const artist = new Artist(artistProps(true, { partner_shows: 1 }))
    expect(artist.initialTabState()).toEqual({ selectedTabIndex: 0, selectedTabTitle: "ABOUT" })
  })
})

describe("layout", () => {
  it("works as expected with no tabs", () => {
    const artist = renderer.create(<Artist artist={artistProps(false).artist} />)
    expect(artist.toJSON()).toMatchSnapshot()
  })

  it("works as expected with one tab", () => {
    const artist = renderer.create(<Artist artist={artistProps(true).artist} />)
    expect(artist.toJSON()).toMatchSnapshot()
  })

  it("works as expected with three tabs", () => {
    const artist = renderer.create(<Artist artist={artistProps(true, { artworks: 2, partner_shows: 1 }).artist} />)
    expect(artist.toJSON()).toMatchSnapshot()
  })
})

const artistProps = (hasMetadata: boolean, counts?: any) => {
  if (!counts) {
    counts = { articles: 0, partner_shows: 0, artworks: 0 }
  }
  return {
    artist: {
      _id: null,
      id: null,
      has_metadata: hasMetadata,
      counts,
      birthday: null,
      name: null,
      nationality: null,
      current_shows: null,
      upcoming_shows: null,
      past_small_shows: null,
      past_large_shows: null,
    },
  }
}
