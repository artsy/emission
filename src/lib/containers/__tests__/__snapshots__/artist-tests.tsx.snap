exports[`layout works as expected with no tabs 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }>
      <Header
        artist={
          Object {
            "counts": Object {
              "articles": 0,
              "artworks": 0,
              "partner_shows": 0,
            },
            "has_metadata": false,
          }
        } />
      <View
        style={
          Object {
            "paddingTop": 30,
          }
        } />
    </View>
  </View>
</RCTScrollView>
`;

exports[`layout works as expected with one tab 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }>
      <Header
        artist={
          Object {
            "counts": Object {
              "articles": 0,
              "artworks": 0,
              "partner_shows": 0,
            },
            "has_metadata": true,
          }
        } />
      <View
        style={
          Object {
            "paddingTop": 30,
          }
        }>
        <About
          artist={
            Object {
              "counts": Object {
                "articles": 0,
                "artworks": 0,
                "partner_shows": 0,
              },
              "has_metadata": true,
            }
          } />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`layout works as expected with three tabs 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "paddingLeft": 40,
          "paddingRight": 40,
        }
      }>
      <Header
        artist={
          Object {
            "counts": Object {
              "artworks": 2,
              "partner_shows": 1,
            },
            "has_metadata": true,
          }
        } />
      <View>
        <SwitchView
          onSelectionChange={[Function]}
          selectedIndex={1}
          style={
            Object {
              "marginBottom": 30,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 30,
            }
          }
          titles={
            Array [
              "ABOUT",
              "WORKS",
              "SHOWS",
            ]
          } />
        <View>
          <Artworks
            artist={
              Object {
                "counts": Object {
                  "artworks": 2,
                  "partner_shows": 1,
                },
                "has_metadata": true,
              }
            } />
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
